__mocks__/dprNewApplicationFactory.ts

```diff
@@ -190,6 +195,44 @@ export const generateAllPossibleDates = (
   return dates;
 };
 
+/**
+ * Generates a random PublicComment
+ *
+ * @returns a randomly generated PublicComment
+ */
+export const generateComment = (numberOfTopics: number = 1) => {
```

Comment:
You're types are returning little strangely from this method, You're also missing tests for this.

if you switch 

```
Omit<PublicComment, "comment"> & {
    comment: TopicAndComments[];
  }
```

to 

```
PublicComment
```

and set the sentiment type

```
faker.helpers.arrayElement<CommentSentiment>([
      "objection",
      "neutral",
      "supportive",
    ])
```

it will return the correct `PublicComment` type